[Bug]:tests target got link error when building with arrow 12.0.0 or greater

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

linking libgar got error:
```
/usr/bin/ld: /workspaces/GraphScope/v6d/build/shared-lib/libgar.so: undefined reference to `arrow::acero::ExecPlan::finished()'
/usr/bin/ld: /workspaces/GraphScope/v6d/build/shared-lib/libgar.so: undefined reference to `arrow::acero::ExecPlan::StopProducing()'
/usr/bin/ld: /workspaces/GraphScope/v6d/build/shared-lib/libgar.so: undefined reference to `arrow::acero::MakeGeneratorReader(std::shared_ptr<arrow::Schema>, std::function<arrow::Future<std::optional<arrow::compute::ExecBatch> > ()>, arrow::MemoryPool*)'
/usr/bin/ld: /workspaces/GraphScope/v6d/build/shared-lib/libgar.so: undefined reference to `arrow::acero::ExecPlan::StartProducing()'
/usr/bin/ld: /workspaces/GraphScope/v6d/build/shared-lib/libgar.so: undefined reference to `arrow::acero::ExecPlan::Make(arrow::compute::ExecContext*, std::shared_ptr<arrow::KeyValueMetadata const>)'
/usr/bin/ld: /workspaces/GraphScope/v6d/build/shared-lib/libgar.so: undefined reference to `arrow::acero::ExecPlan::Validate()'
/usr/bin/ld: /workspaces/GraphScope/v6d/build/shared-lib/libgar.so: undefined reference to `arrow::acero::default_exec_factory_registry()'
collect2: error: ld returned 1 exit status
make[2]: *** [modules/graph/CMakeFiles/vineyard-graph-loader.dir/build.make:155: bin/vineyard-graph-loader] Error 1
make[1]: *** [CMakeFiles/Makefile2:1349: modules/graph/CMakeFiles/vineyard-graph-loader.dir/all] Error 2
```

### Expected Behavior

got no error

### Minimal Reproducible Example

install arrow 12.0.0 and building GraphAr cpp with test

### Environment

- Operating system: ubuntu22.04
- GraphAr version: v0.5.0


### Link to GraphAr Logs

_No response_

### Further Information

_No response_

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -50,9 +50,9 @@ jobs:
         # install the latest arrow deb to test arrow
         wget -c https://apache.jfrog.io/artifactory/arrow/"$(lsb_release --id --short | tr 'A-Z' 'a-z')"/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb \
             -P /tmp/
-        apt-get install -y -V /tmp/apache-arrow-apt-source-latest-"$(lsb_release --codename --short)".deb
+        sudo apt-get install -y -V /tmp/apache-arrow-apt-source-latest-"$(lsb_release --codename --short)".deb
         sudo apt-get update -y
-        apt-get install -y libarrow-dev
+        sudo apt-get install -y libarrow-dev
         sudo apt-get install -y libboost-graph-dev ccache libcurl4-openssl-dev
 
     - name: CMake

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -46,7 +46,13 @@ jobs:
 
     - name: Install dependencies
       run: |
+
+        # install the latest arrow deb to test arrow
+        wget -c https://apache.jfrog.io/artifactory/arrow/"$(lsb_release --id --short | tr 'A-Z' 'a-z')"/apache-arrow-apt-source-latest-$(lsb_release --codename --short).deb \
+            -P /tmp/
+        apt-get install -y -V /tmp/apache-arrow-apt-source-latest-"$(lsb_release --codename --short)".deb
         sudo apt-get update -y
+        apt-get install -y libarrow-dev
         sudo apt-get install -y libboost-graph-dev ccache libcurl4-openssl-dev
 
     - name: CMake

--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -191,10 +191,12 @@ macro(build_gar)
     if(APPLE)
         target_link_libraries(gar PRIVATE -Wl,-force_load gar_arrow_static
             "${GAR_PARQUET_STATIC_LIB}"
+            "${GAR_ACERO_STATIC_LIB}"
             "${GAR_ARROW_BUNDLED_DEPS_STATIC_LIB}")
     else()
         target_link_libraries(gar PRIVATE -Wl,--exclude-libs,ALL -Wl,--whole-archive gar_arrow_static
             "${GAR_PARQUET_STATIC_LIB}"
+            "${GAR_ARROW_ACERO_STATIC_LIB}"
             "${GAR_ARROW_BUNDLED_DEPS_STATIC_LIB}" -Wl,--no-whole-archive)
     endif()
 

--- a/cpp/cmake/apache-arrow.cmake
+++ b/cpp/cmake/apache-arrow.cmake
@@ -33,14 +33,6 @@ function(build_arrow)
         message(SEND_ERROR "Error: unrecognized arguments: ${ARG_UNPARSED_ARGUMENTS}")
     endif ()
 
-    find_package(Threads)
-    find_package(Arrow QUIET)
-    set(ARROW_VERSION_TO_BUILD "10.0.1" CACHE INTERNAL "arrow version")
-    if (Arrow_FOUND) # arrow is installed, build the same version as the installed one
-        message(STATUS "Found Arrow installed, align to version: ${Arrow_VERSION}")
-        set(ARROW_VERSION_TO_BUILD "${Arrow_VERSION}" CACHE INTERNAL "arrow version")
-    endif ()
-
     # If Arrow needs to be built, the default location will be within the build tree.
     set(GAR_ARROW_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/arrow_ep-prefix")
 
@@ -78,6 +70,7 @@ function(build_arrow)
                              "-DBoost_SOURCE=BUNDLED"
                              "-DARROW_ORC=ON"
                              "-DARROW_COMPUTE=ON"
+                             "-DARROW_ACERO=ON"
                              "-DARROW_DATASET=ON"
                              "-DARROW_JEMALLOC=OFF"
                              "-DARROW_WITH_SNAPPY=OFF"
@@ -91,6 +84,14 @@ function(build_arrow)
 
     set(GAR_ARROW_INCLUDE_DIR "${GAR_ARROW_PREFIX}/include" CACHE INTERNAL "arrow include directory")
     set(GAR_ARROW_BUILD_BYPRODUCTS "${GAR_ARROW_STATIC_LIB}" "${GAR_PARQUET_STATIC_LIB}")
+
+    find_package(Threads)
+    find_package(Arrow QUIET)
+    set(ARROW_VERSION_TO_BUILD "10.0.1" CACHE INTERNAL "arrow version")
+    if (Arrow_FOUND) # arrow is installed, build the same version as the installed one
+        message(STATUS "Found Arrow installed, align to version: ${Arrow_VERSION}")
+        set(ARROW_VERSION_TO_BUILD "${Arrow_VERSION}" CACHE INTERNAL "arrow version")
+    endif ()
     set(GAR_ARROW_SOURCE_FILE "https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${ARROW_VERSION_TO_BUILD}/apache-arrow-${ARROW_VERSION_TO_BUILD}.tar.gz")
 
     include(ExternalProject)
@@ -113,6 +114,16 @@ function(build_arrow)
     set_target_properties(${GAR_PARQUET_LIBRARY_TARGET}
             PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${GAR_ARROW_INCLUDE_DIR}
             IMPORTED_LOCATION ${GAR_PARQUET_STATIC_LIB})
+    if (ARROW_VERSION_TO_BUILD GREATER_EQUAL "12.0.0")
+        set(GAR_ARROW_ACERO_STATIC_LIB_FILENAME
+	    "${CMAKE_STATIC_LIBRARY_PREFIX}arrow_acero${CMAKE_STATIC_LIBRARY_SUFFIX}")
+        set(GAR_ARROW_ACERO_STATIC_LIB "${GAR_ARROW_STATIC_LIBRARY_DIR}/${GAR_ARROW_ACERO_STATIC_LIB_FILENAME}" CACHE INTERNAL "acero lib")
+        set(GAR_ARROW_ACERO_LIBRARY_TARGET gar_acero_static)
+        add_library(${GAR_ARROW_ACERO_LIBRARY_TARGET} STATIC IMPORTED)
+        set_target_properties(${GAR_ARROW_ACERO_LIBRARY_TARGET}
+            PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${GAR_ARROW_INCLUDE_DIR}
+            IMPORTED_LOCATION ${GAR_ARROW_ACERO_STATIC_LIB})
+    endif()
 
     add_dependencies(${GAR_ARROW_LIBRARY_TARGET} arrow_ep)
 endfunction()

